VirtoCommerce.Domain/Pricing/Services/IPricingService.cs

```diff
@@ -19,5 +20,8 @@ public interface IPricingService
 
         IEnumerable<Pricelist> EvaluatePriceLists(PriceEvaluationContext evalContext);
         IEnumerable<Price> EvaluateProductPrices(PriceEvaluationContext evalContext);
+
+        IEnumerable<Model.PriceCalendarChange> GetCalendarChanges(DateTime? lastEvaluationTimestamp,
```

Comment:
This method in the IPricingService also looks a redundant. These logic can be implemented inside `ProductPriceDocumentChangesProvider` with   injection and direct access to `IPricingRepository` in the logic.

Reply:
We need it somewhere else as well, because we index based on events and not based on the timer.
If you want we can move it the the IPricingRepository, but I don't like to depend on these repositories because they expose IQueryable which is a quite a leaky abstraction.

Reply:
Refactored